File: SDKs/zuora.com/2021-08-20/typescript/src/sdk/models/operations/getqueryemailtemplates.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetQueryEmailTemplatesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=eventTypeName" })
  eventTypeName?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=start" })
  start?: number;
}


export class GetQueryEmailTemplatesHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Authorization" })
  authorization: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Zuora-Entity-Ids" })
  zuoraEntityIds?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Zuora-Track-Id" })
  zuoraTrackId?: string;
}


export class GetQueryEmailTemplates200ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data", elemType: shared.GetPublicEmailTemplateResponse })
  data?: shared.GetPublicEmailTemplateResponse[];

  @SpeakeasyMetadata({ data: "json, name=next" })
  next?: string;
}


export class GetQueryEmailTemplatesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetQueryEmailTemplatesQueryParams;

  @SpeakeasyMetadata()
  headers: GetQueryEmailTemplatesHeaders;
}


export class GetQueryEmailTemplatesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  errorResponse?: shared.ErrorResponse;

  @SpeakeasyMetadata()
  getQueryEmailTemplates200ApplicationJSONObject?: GetQueryEmailTemplates200ApplicationJson;

  @SpeakeasyMetadata()
  headers: Record<string, string[]>;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/zuora.com/2021-08-20/python/src/sdk/models/operations/get_query_email_templates.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import getpublicemailtemplateresponse as shared_getpublicemailtemplateresponse
from ..shared import errorresponse as shared_errorresponse


@dataclasses.dataclass
class GetQueryEmailTemplatesQueryParams:
    event_type_name: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'eventTypeName', 'style': 'form', 'explode': True }})
    limit: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'limit', 'style': 'form', 'explode': True }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'name', 'style': 'form', 'explode': True }})
    start: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'start', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class GetQueryEmailTemplatesHeaders:
    authorization: str = dataclasses.field(metadata={'header': { 'field_name': 'Authorization', 'style': 'simple', 'explode': False }})
    zuora_entity_ids: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'Zuora-Entity-Ids', 'style': 'simple', 'explode': False }})
    zuora_track_id: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'Zuora-Track-Id', 'style': 'simple', 'explode': False }})
    

@dataclass_json
@dataclasses.dataclass
class GetQueryEmailTemplates200ApplicationJSON:
    data: Optional[list[shared_getpublicemailtemplateresponse.GetPublicEmailTemplateResponse]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    next: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next') }})
    

@dataclasses.dataclass
class GetQueryEmailTemplatesRequest:
    headers: GetQueryEmailTemplatesHeaders = dataclasses.field()
    query_params: GetQueryEmailTemplatesQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class GetQueryEmailTemplatesResponse:
    content_type: str = dataclasses.field()
    headers: dict[str, list[str]] = dataclasses.field()
    status_code: int = dataclasses.field()
    error_response: Optional[shared_errorresponse.ErrorResponse] = dataclasses.field(default=None)
    get_query_email_templates_200_application_json_object: Optional[GetQueryEmailTemplates200ApplicationJSON] = dataclasses.field(default=None)
    
